packages/core/src/utils/validateLocaleForHostname.ts

```diff
@@ -0,0 +1,66 @@
+import localesSettings from '../../locales-settings.json'
+import type { LocalesSettings } from '../typings/locales'
+
+// TODO: update to use SDK
+function getLocalesSettings(): LocalesSettings {
+  return localesSettings as LocalesSettings
+}
+
+/**
+ * A binding is valid if the hostname matches AND the locale code matches.
+ * This works for both:
+ * - Path-based bindings (e.g., "https://brandless.fast.store/pt-BR")
+ * - Subdomain bindings (e.g., "https://canada.brandless.fast.store" for "fr-CA")
+ *
+ * @param hostname - The request hostname
+ * @param localeCode - The locale code to check
+ * @returns true if there's a valid binding for this hostname and locale, false otherwise
+ */
+function hasValidBindingForHostname(
+  hostname: string,
+  localeCode: string
+): boolean {
+  const localesSettings = getLocalesSettings()
+  const localeData = localesSettings.locales[localeCode]
+
+  if (!localeData?.bindings) {
+    return false
+  }
+
+  return localeData.bindings.some((binding) => {
+    if (!binding.url) return false
```

Comment:
Must every binding have a URL? 
If we have 2 bindings, the first without a URL and the second matches the URL hostname, is this not valid?